libs/sdk-ui-ext/src/internal/dashboardEmbedding/hooks/useExportDashboardToPdf.ts

```diff
@@ -0,0 +1,94 @@
+// (C) 2020-2021 GoodData Corporation
+import { AnalyticalBackendError, IAnalyticalBackend } from "@gooddata/sdk-backend-spi";
+import { ObjRef } from "@gooddata/sdk-model";
+import {
+    useBackend,
+    useCancelablePromise,
+    UseCancelablePromiseCallbacks,
+    UseCancelablePromiseState,
+    useWorkspace,
+} from "@gooddata/sdk-ui";
+import invariant from "ts-invariant";
+import { IDashboardFilter } from "../types";
+import { dashboardFilterToFilterContextItem } from "../utils/filters";
+
+/**
+ * @beta
+ */
+export interface IUseExportDashboardToPdfConfig
+    extends UseCancelablePromiseCallbacks<string, AnalyticalBackendError> {
+    /**
+     * Reference to the dashboard to export.
+     */
+    dashboard?: ObjRef;
+
+    /**
+     * Optionally, specify filters to be applied to all the widgets in the dashboard.
+     * This will override any filters set on the dashboard itself.
+     */
+    filters?: IDashboardFilter[];
+
+    /**
+     * Backend to work with.
+     *
+     * Note: the backend must come either from this property or from BackendContext. If you do not specify
+     * backend here, then the hook MUST be called within an existing BackendContext.
+     */
+    backend?: IAnalyticalBackend;
+
+    /**
+     * Workspace to work with.
+     *
+     * Note: the workspace must come either from this property or from WorkspaceContext. If you do not specify
+     * workspace here, then the hook MUST be called within an existing WorkspaceContext.
+     */
+    workspace?: string;
+}
+
+/**
+ * Hook allowing to export a dashboard to a PDF
+ * @param config - configuration of the hook
+ * @beta
+ */
+export function useExportDashboardToPdf({
```

Comment:
there are two hooks `useExportDashboardToPdf`. this one and another in `convenience` dir. the one from `convenience` is exported...

would it make sense to rename this function so that we don't have two hooks with same name?

Reply:
Good point, renamed.